### PR TITLE
Contract tests

### LINKED ISSUE #273: Add test for contract/contractFactory utilities
Like `contractFactory.getClassHash`, or `contract.getAbi`, these aren't tested anywhere

### PR DESCRIPTION
**Pull request by shramee**
Fixes #273

## Usage related changes

None

## Development related changes

Additional tests added for `StarknetContract` and `StarknetContractFactory`.

#### `function getPredeployedAccount()`
Returns `{Promise<PredeployedAccount>}`  details of a pre-deployed account.

#### `function getPredeployedOZAccount()`
Returns `{Promise<OpenZeppelinAccount>}` An OZAccount instance for a pre-deployed account.

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Documented the changes
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.ts`)
-   [x] Linked issues which this PR resolves
-   [x] Created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example):
    -   https://github.com/Shard-Labs/starknet-hardhat-example/pull/92
    -   [ ] Modified `test.sh` to use my example repo branch
    -   [ ] Restored `test.sh` to to use the original branch (after the example repo PR has been merged)
-   [ ] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: test/general-tests/contract-test/hardhat.config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = {
+    starknet: {
+        network: process.env.NETWORK
+    },
+    networks: {
+        integratedDevnet: {
+            url: "http://127.0.0.1:5050",
+            // testing with fork because alpha-goerli has the needed argent account contracts declared
+            // using integrated-devnet (in network.json) because spawning devnet is currently out of reach for invidiual tests
```
[the remark this answers is not in the document]
Comment (shramee) [reply test/general-tests/contract-test/hardhat.config.ts]: Oh, fixed

File: test/general-tests/contract-test/hardhat.config.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import "@shardlabs/starknet-hardhat-plugin";
+
+module.exports = {
+    starknet: {
+        network: process.env.NETWORK
+    },
+    networks: {
+        integratedDevnet: {
+            url: "http://127.0.0.1:5050",
+            // testing with fork because alpha-goerli has the needed argent account contracts declared
```
[the remark this answers is not in the document]
Comment (shramee) [reply test/general-tests/contract-test/hardhat.config.ts]: Removed the comment, seems unnecessary.

**Review by FabijanC — APPROVED** [review]

**Comment by shramee** [comment]
Yeah sure

**Comment by shramee** [comment]
@FabijanC Addressed in https://github.com/Shard-Labs/starknet-hardhat-example/pull/94

**Pull request merged**